types/blob/blob.go

```diff
@@ -0,0 +1,67 @@
+package blob
+
+import (
+	"bytes"
+	"encoding/json"
+
+	"github.com/rollkit/celestia-openrpc/types/namespace"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+// Commitment is a Merkle Root of the subtree built from shares of the Blob.
+// It is computed by splitting the blob into shares and building the Merkle subtree to be included
+// after Submit.
+type Commitment []byte
+
+func (com Commitment) String() string {
+	return string(com)
+}
+
+// Equal ensures that commitments are the same
+func (com Commitment) Equal(c Commitment) bool {
+	return bytes.Equal(com, c)
+}
+
+// Blob represents any application-specific binary data that anyone can submit to Celestia.
+type Blob struct {
+	tmproto.Blob `json:"blob"`
```

Comment:
why do you need `Blob` in tendermint proto? can you not just do it in rollkit.proto? that way, cometbft need not be changed. also feels `Blob` is something rollkit specific and not tendermint/cometbft

Reply:
I need it in `celestia-openrpc` as well to match the client signature: https://github.com/rollkit/celestia-openrpc/pull/11/files#diff-f7bb71ca856c07b3329265f26b1e577e00b71e4792550cf14346bdbe0e7bd18dR74

Adding it to rollkit would create a circular dependency.

Reply:
It was added to `celestia-core` here: https://github.com/celestiaorg/celestia-core/blob/8bbae5a6b04d5f2ad9ede83233ecfbdc4d33231e/proto/tendermint/types/types.proto#L106-L115

Reply:
can the Blob be added to celestia-openrpc? then you won't have the circular dependency right? i am hesitant to add this to cometbft because it is specific to celestia and not other DAs.

Reply:
Yeah it would add some small overhead in terms of proto boiler code, but that's fine.